lib/middleware/verify_hideout_id.rb

```diff
@@ -0,0 +1,39 @@
+require 'json'
+
+require_relative '../../app/helpers/auth_helper.rb'
+
+module Middleware
+    class VerifyHideoutId
+      def initialize(app)
+        @app = app
+      end
+  
+      def call(env)
+        request = ActionDispatch::Request.new(env)
+        request_identifier = "#{request.method} #{request.path}"
+        if not ENV['PUBLIC_ROUTES'].include?(request_identifier)
+          authoriation_header = request.headers[:Authorization]
+          return [401, {}, []] if authoriation_header.nil?
+  
+          access_token = authoriation_header.split[1]
+          result = AuthHelper.validate_token_by_type(:ACCESS, access_token)
+          return [401, {}, []] unless result[:success]
+  
+          begin
+            payload = result[:payload]
+            payload.transform_keys!(&:to_sym)
+            current_hideout = Hideout.find_by(id: payload[:hideout_id])
+            return [400, {}, []] unless current_hideout == request.params[:hideout_id].to_i
```

Comment:
Ok this is the updated comment. Just check two things: 1. the hideout id passed in the url is the same as the one in `request[:payload][:hideout_id]` and 2. that this id is an actual hideout using `find_by!` and the rescue that you currently have.

Reply:
I have made 2 different verify_hideout_id files one of them has the old requested changes and the new one has this change so if you want additional changes I can make them and just let me know how you want it to be finally